### PR TITLE
persist controller: sub-persist controllers registration and  implementations.

### PR DESCRIPTION
**Pull request by SimonCqk**
FEATURE: watch jobs/pods/events and persist them into storage backends by provided core-api, so that users of kubedl will have the abilities to look up meta-data and states of those one who have been deleted from etcd.

### CODE REVIEW

File: controllers/persist/event/events_event_handler.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+var _ handler.EventHandler = &enqueueForEvent{}
+
+// enqueueForEvent implements EventHandler interface and filter out events
+// not provided by kubedl managed objects.
+type enqueueForEvent struct{}
+
+func (e *enqueueForEvent) Create(evt event.CreateEvent, queue workqueue.RateLimitingInterface) {
+	evtObj := evt.Object.(*corev1.Event)
+	if !util.IsKubeDLManagedObjectKind(evtObj.InvolvedObject.Kind) {
```
Comment (tzzcfrank) [inline controllers/persist/event/events_event_handler.go:38]: the kind of pod must be considered in this condition

File: controllers/persist/persist_controller.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		if err != nil {
+			return err
+		}
+		if err = eventPersistController.SetupWithManager(mgr); err != nil {
+			return err
+		}
+	}
+
+	if objectStorage != "" {
+		println("1")
```
Comment (tzzcfrank) [inline controllers/persist/persist_controller.go:58]: remove debug code

Comment (SimonCqk) [reply controllers/persist/persist_controller.go]: done

File: controllers/persist/event/events_event_handler.go (hunk 1 above)
Comment (SimonCqk) [reply controllers/persist/event/events_event_handler.go]: done

**Comment by tzzcfrank** [comment]
LGTM

**Pull request merged**